src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -892,37 +935,54 @@ def is_frontend_port_present_in_host():
     return True
 
 
-def get_num_dpus():
+def get_dpu_info():
     """
-    Retrieves the number of DPUs from platform.json file.
-
-    Args:
+    Retrieves the DPU information from platform.json file.
 
     Returns:
-        A integer to indicate the number of DPUs.
+        A dictionary containing the DPU information.
     """
 
     platform = get_platform()
     if not platform:
-        return 0
+        return {}
 
-    # Get Platform path.
-    platform_path = get_path_to_platform_dir()
+    # Retrieve platform.json data
+    platform_data = get_platform_json_data()
+    if platform_data:
+        # Convert keys to lower case to avoid case sensitivity.
+        data = {k.lower(): v for k, v in platform_data.items()}
+        dpu_info = data.get('dpus', {})
+        return dpu_info
 
-    if os.path.isfile(os.path.join(platform_path, PLATFORM_JSON_FILE)):
-        json_file = os.path.join(platform_path, PLATFORM_JSON_FILE)
+    return {}
 
-        try:
-            with open(json_file, 'r') as file:
-                platform_data = json.load(file)
-        except (json.JSONDecodeError, IOError, TypeError, ValueError):
-            # Handle any file reading and JSON parsing errors
-            return 0
 
-        # Convert to lower case avoid case sensitive.
-        data = {k.lower(): v for k, v in platform_data.items()}
-        DPUs = data.get('dpus', None)
-        if DPUs is not None and len(DPUs) > 0:
-            return len(DPUs)
+def get_num_dpus():
+    """
+    Retrieves the number of DPUs from platform.json file.
+
+    Returns:
+        A integer to indicate the number of DPUs.
+    """
+
+    dpu_info = get_dpu_info()
+    if dpu_info is not None and len(dpu_info) > 0:
+        return len(dpu_info)
 
     return 0
+
+
+def get_dpu_list():
+    """
+    Retrieves the list of DPUs from platform.json file.
```

Comment:
Why is this needed? What is the difference from `get_dpu_info`?

Reply:
get_dpu_list() only returns the list of DPU module names on NPU, whereas get_dpu_info() returns both the DPU module information and the values for all the DPU modules.

Reply:
For example, if platform.json is 
"{
    "DPUS": {
        "dpu0": {
            "midplane_interface": "dpu0"
        },
        "dpu1": {
            "midplane_interface": "dpu1"
        }
    }
}"

get_dpu_info() returns
{
    "dpu0": {
        "midplane_interface": "dpu0"
    },
    "dpu1": {
        "midplane_interface": "dpu1"
    }
}

and get_dpu_list() return this
["dpu0", "dpu1"]

Reply:
@vvolam can you specify the return accordingly in the code comment?

Reply:
Added an example. Thank you!